src/artifact-utils.ts

```diff
@@ -30,13 +30,15 @@ function mirrorVar(
   defaultValue: string,
 ): string {
   // Convert camelCase to camel_case for env var reading
-  const lowerName = name.replace(/([a-z])([A-Z])/g, (_, a, b) => `${a}_${b}`).toLowerCase();
+  const snakeName = name.replace(/([a-z])([A-Z])/g, (_, a, b) => `${a}_${b}`).toLowerCase();
 
   return (
-    process.env[`NPM_CONFIG_ELECTRON_${lowerName.toUpperCase()}`] ||
-    process.env[`npm_config_electron_${lowerName}`] ||
-    process.env[`npm_package_config_electron_${lowerName}`] ||
-    process.env[`ELECTRON_${lowerName.toUpperCase()}`] ||
+    // configs coming from .npmrc are always lowercase regardless how it was defined
+    process.env[`npm_config_electron_${name.toLowerCase()}`] ||
+    process.env[`NPM_CONFIG_ELECTRON_${snakeName.toUpperCase()}`] ||
+    process.env[`npm_config_electron_${snakeName}`] ||
+    process.env[`npm_package_config_electron_${name}`] ||
+    process.env[`ELECTRON_${snakeName.toUpperCase()}`] ||
     options[name] ||
     defaultValue
   );
```

Comment:
This is pretty much (almost) covered by tests already.
Change only adds support for more variations in the env variables

| Env Variable | Old | New |
|:-|:-|:-|
|***.npmrc***|
|`npm_config_electron_nighlymirror`|❌|✅|
|`NPM_CONFIG_ELECTRON_NIGHLY_MIRROR`|✅|✅|
|`npm_config_electron_nighly_mirror`|✅|✅|
|***package.json***|
|`npm_package_config_electron_nighlyMirror`|❌|✅|
|`npm_package_config_electron_nighly_mirror`|✅|✅|
|***Global***|
|`ELECTRON_NIGHLY_MIRROR`|✅|✅|